### PR TITLE
Remove magic bucket naming and validate on bucket name create

### PR DESCRIPTION
**Pull request by jmoz**
## What

Remove magic bucket naming and validate on bucket name create

## How to review

1. Run tests

https://trello.com/c/WWG3HLi2/386-bucket-naming-should-be-validated-on-creation-as-such-dev-bucketname-and-server-check-for-correct-env

### CODE REVIEW

File: control_panel_api/models.py [hunk 1]
```diff
@@ -32,24 +31,10 @@ class Meta:
 
 
 class S3Bucket(TimeStampedModel):
-    # See: AWS' Bucket Restrictions and Limitations
```
Comment (xoen) [inline control_panel_api/models.py]: I think I like it more with the validations in a separate module 👍

File: control_panel_api/models.py [hunk 2]
```diff
@@ -32,24 +31,10 @@ class Meta:
...
-    LENGTH_REGEX = '^.{3,63}$'
-
     name = models.CharField(unique=True, max_length=63, validators=[
-        RegexValidator(regex=LENGTH_REGEX,
-                       message="must be between 3 and 63 characters"),
-        RegexValidator(regex=LABELS_REGEX,
-                       message="is invalid, check AWS S3 bucket names restrictions (for example, can only contains letters, digits, dots and hyphens)"),
+        validators.validate_env_prefix,
+        validators.validate_s3_bucket_length,
+        validators.validate_s3_bucket_label,
```
Comment (xoen) [inline control_panel_api/models.py]: I know the name of this wasn't probably great in the first place but I'd probably call this `validators.validate_s3_bucket_labels` (plural) as the idea was for this regex to validate that the S3 bucket name was 1+ labels (as defined by AWS S3 bucket naming restrictions document).

File: control_panel_api/tests/test_views.py [hunk 3]
```diff
@@ -239,11 +241,27 @@ def test_create_when_name_invalid(self):
...
+        data = {'name': 'bucketname'}
+        response = self.client.post(reverse('s3bucket-list'), data)
+        self.assertEqual(HTTP_400_BAD_REQUEST, response.status_code)
+
     @patch('control_panel_api.services.bucket_create')
     def test_create_calls_apis(self, mock_bucket_create):
         data = {'name': 'test-bucket-123'}
         self.client.post(reverse('s3bucket-list'), data)
-        mock_bucket_create.assert_called()
+        mock_bucket_create.assert_called_with('test-bucket-123')
```
Comment (xoen) [inline control_panel_api/tests/test_views.py]: better 👍

**Comment by xoen** [comment]
Looks good to me 👍

File: control_panel_api/models.py (hunk 2 above)
Comment (jmoz) [reply control_panel_api/models.py]: Done

**Pull request merged**